cpp/bindings/CMakeLists.txt

```diff
@@ -11,11 +11,21 @@ if(${PYTHON_BINDINGS})
   # Build bindings ###################################################
   find_package(pybind11 REQUIRED)
   message(STATUS "Found ${PYTHON_INCLUDE_DIR} ${PYTHON_EXECUTABLE}")
-  pybind11_add_module(toppra_int toppra_int.cpp bindings.cpp)
+  pybind11_add_module(toppra_int toppra_int.cpp bindings.cpp
+    constraints.cpp paths.cpp)
   target_link_libraries(toppra_int PUBLIC toppra)
   target_include_directories(toppra_int PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
   set_target_properties(toppra_int PROPERTIES
     LIBRARY_OUTPUT_DIRECTORY ${TOPPRA_PYTHON_SOURCE_DIR}
     )
+  if(BUILD_WITH_PINOCCHIO)
+    target_compile_options(toppra_int PRIVATE -DBUILD_WITH_PINOCCHIO)
+    target_link_libraries(toppra_int PUBLIC pinocchio::pinocchio)
+    if(DEFINED PINOCCHIO_WITH_PYTHON_INTERFACE)
+      find_package(Boost REQUIRED COMPONENTS python3)
+      target_compile_options(toppra_int PRIVATE -DPINOCCHIO_WITH_PYTHON_INTERFACE)
+      target_link_libraries(toppra_int PUBLIC Boost::python3)
```

Comment:
I don't see how PINOCCHIO_WITH_PYTHON_INTERFACE is defined?

Also, can we find different boost component based on the PYTHON_VERSION option?

Reply:
It is defined in the `pinocchioConfig.cmake`, included when doing `find_package(pinocchio)`.

Reply:
For Python 3, I think you can use component `python3-py36` for Python 3.6 but I am not sure.

Reply:
I tried to find pinocchio for python3 on robotpkg, there is only the 3.6 binding. Do you say it's easy/possible to build it from source for a different version (i.e. 3.7)?

Reply:
Should be relatively easy:
```bash
sudo apt remove robotpkg-pinocchio
# (optional, for collision detection) sudo apt install robotpkg-hpp-fcl 
git clone --recursive https://github.com/stack-of-tasks/pinocchio
mkdir build && cd build
cmake -DPYTHON_EXECUTABLE=<path to python 3.7 interpreter> ..
# ccmake .. to tune the few compilation options
make install # go take a coffee, it's long and memory consuming.
```

Reply:
You can also install Pinocchio from anaconda or pip.